src/Core.UnitTests/Helpers/PathHelperTests.cs

```diff
@@ -385,6 +385,34 @@ public void GetTempDirForTask_perVSInstanceTrueHasPath_AddsGuid()
             Guid.TryParse(actualPath.Split('\\').Last(), out _).Should().BeTrue();
         }
 
+        [TestMethod]
+        // Module-level issues i.e. no file
+        [DataRow(null, null, true)]
+        [DataRow(null, "", true)]
+        [DataRow("", null, true)]
+        [DataRow("", "", true)]
+
+        // Module-level issues should not match non-module-level issues
+        [DataRow(@"any.txt", "", false)]
+        [DataRow(@"any.txt", null, false)]
+        [DataRow("", @"c:\any.txt", false)]
+        [DataRow(null, @"c:\any.txt", false)]
+
+        // File issues
+        [DataRow(@"same.txt", @"c:\same.txt", true)]
+        [DataRow(@"SAME.TXT", @"c:\same.txt", true)]
+        [DataRow(@"same.TXT", @"c:\XXXsame.txt", false)]  // partial file name -> should not match
+        [DataRow(@"differentExt.123", @"a:\differentExt.999", false)] // different extension -> should not match
+        [DataRow(@"aaa\partial\file.cs", @"d:\partial\file.cs", false)]
+        // Only matching the local path tail, so the same server path can match multiple local files
+        [DataRow(@"partial\file.cs", @"c:\aaa\partial\file.cs", true)]
+        [DataRow(@"partial\file.cs", @"c:\aaa\bbb\partial\file.cs", true)]
+        [DataRow(@"partial\file.cs", @"c:\aaa\bbb\ccc\partial\file.cs", true)]
+        public void IsFileMatch_ReturnsExpectedResult(string serverFilePath, string localFilePath, bool expected)
+        {
+            // Act and assert
+            PathHelper.IsServerFileMatch(localFilePath, serverFilePath).Should().Be(expected);
+        }
```

Comment:
The test inputs are the same the corresponding `SuppressedIssueMatcher` test.
This duplication isn't ideal, but the alternatives are:
1) drop the corresponding `SuppressedIssueMatcher` test, - bad, as we'd have no coverage of this case.
2) refactor so we could inject an `IPathHelper` into the `SuppressedIssueMatcher`. This is the goal in the long run, but it's too much work to do at the moment.

@rita-sonarsource and I discussed the options yesterday and the current plan is:
* move the file-checking logic to `PathHelper` (i.e. this commit)
* change the Roslyn suppression checker to use the new `PathHelper` method (which will fix bug #3997)
* create a ticket to create an `IPathHelper` and refactor the existing code and tests.

In the meantime we'll live with the duplication in the test code.